.github/actions/common-setup/action.yml

```diff
@@ -0,0 +1,23 @@
+name: Setup Environment
+inputs:
+  CACHIX_SIGNING_KEY: # determines what node version to install
```

Comment:
I don't understand this comment. Why not use `secrets.CACHIX_SIGNING_KEY` directly?

Reply:
You can't access secrets from the action.
Also the checkout cant be factored out into the action as the action file has to be present before gh can call it.